File: Microsoft.NET/SampleShopApplication/SampleShopApplication/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using System.Threading.Tasks;
using YenePaySdk;

namespace SampleShopApplication.Controllers
{
    public class HomeController : Controller
    {
        private CheckoutOptions checkoutoptions;
        private string pdtToken = "YOUR_PDT_KEY_HERE";

        public HomeController()
        {
            string sellerCode = "YOUR_YENEPAY_SELLER_CODE";
            string successUrlReturn = "http://localhost:5525/Home/PaymentSuccessReturnUrl"; //"YOUR_SUCCESS_URL";
            string ipnUrlReturn = "http://localhost:5525/Home/IPNDestination"; //"YOUR_IPN_URL";
            string cancelUrlReturn = ""; //"YOUR_CANCEL_URL";
            string failureUrlReturn = ""; //"YOUR_FAILURE_URL";
            bool useSandBox = true;
            checkoutoptions = new CheckoutOptions(sellerCode, string.Empty, CheckoutType.Express, useSandBox, null, successUrlReturn, cancelUrlReturn, ipnUrlReturn, failureUrlReturn);
        }

        //
        // GET: /Home/
        public ActionResult Index()
        {
            return View();
        }

        /// <summary>
        /// Handles the Express checkout type which allows buying a single item from your site or online store
        /// </summary>
        [HttpPost]
        public void CheckoutExpress()
        {
            checkoutoptions.Process = CheckoutType.Express;
            var itemId = Request.Form["ItemId"];
            var itemName = Request.Form["ItemName"];
            var unitPrice = decimal.Parse(Request.Form["UnitPrice"]);
            var quantity = int.Parse(Request.Form["Quantity"]);
            var discount = decimal.Parse(Request.Form["Discount"]);
            var deliveryFee = decimal.Parse(Request.Form["DeliveryFee"]);
            var handlingFee = decimal.Parse(Request.Form["HandlingFee"]);
            var tax1 = decimal.Parse(Request.Form["Tax1"]);
            var tax2 = decimal.Parse(Request.Form["Tax2"]);

            CheckoutItem checkoutitem = new CheckoutItem(itemId, itemName, unitPrice, quantity, tax1, tax2, discount, handlingFee, deliveryFee);
            checkoutoptions.OrderId = "12-34"; //"YOUR_UNIQUE_ID_FOR_THIS_ORDER";  //can also be set null
            checkoutoptions.ExpiresInDays = 2; //"NUMBER_OF_DAYS_BEFORE_THE_ORDER_EXPIRES"; //setting null means it never expires
            var url = CheckoutHelper.GetCheckoutUrl(checkoutoptions, checkoutitem);
            Response.Redirect(url);
        }

        [HttpGet]
        public ActionResult Cart()
        {
            return View();
        }

        /// <summary>
        /// Handles the Cart checkout type which allows buying multiple items at once from your site or online store
        /// </summary>
        /// <param name="Items">A list of Items selected by your customer</param>
        /// <returns>A JSON object containing the checkout payment URL which you can use to redirect a customer to YenePay's payment processing site</returns>
        [HttpPost]
        public ActionResult CheckoutCart(List<CheckoutItem> Items)
        {
            checkoutoptions.Process = CheckoutType.Cart;
            decimal? totalItemsDeliveryFee = 50;
            decimal? totalItemsDiscount = 30;
            decimal? totalItemsHandlingFee = 10;
            decimal? totalItemsTax1 = Items.Sum(i=> (i.UnitPrice*i.Quantity)) * (decimal)0.15;
            decimal? totalItemsTax2 = Items.Sum(i => (i.UnitPrice * i.Quantity)) * (decimal)0.02;
            checkoutoptions.SetOrderFees(totalItemsDeliveryFee, totalItemsDiscount, totalItemsHandlingFee, totalItemsTax1, totalItemsTax2);

            checkoutoptions.OrderId = "AB-CD"; //"YOUR_UNIQUE_ID_FOR_THIS_ORDER";  //can also be set null
            checkoutoptions.ExpiresInDays = 2; //"NUMBER_OF_DAYS_BEFORE_THE_ORDER_EXPIRES"; //setting null means it never expires

            var url = CheckoutHelper.GetCheckoutUrl(checkoutoptions, Items);
            return Json(new { redirectUrl = url });

        }

        [HttpPost]
        public async Task<string> IPNDestination(IPNModel ipnModel)
        {
            var result = string.Empty;
            ipnModel.UseSandbox = checkoutoptions.UseSandbox;
            if (ipnModel != null)
            {
                var isIPNValid = await CheckIPN(ipnModel);

                if (isIPNValid)
                {
                    //This means the payment is completed
                    //You can now mark the order as "Paid" or "Completed" here and start the delivery process
                }
            }
            return result;
        }

        public async Task<ActionResult> PaymentSuccessReturnUrl(IPNModel ipnModel)
        {
            Guid transactionId = ipnModel.TransactionId;
            PDTRequestModel model = new PDTRequestModel(pdtToken, transactionId);
            var pdtString = await CheckoutHelper.RequestPDT(model);
            if (!string.IsNullOrEmpty(pdtString))
            {
                //This means the payment is completed. 
                //You can extract more information of the transaction from the pdtString
                //You can now mark the order as "Paid" or "Completed" here and start the delivery process
            }
            else {
                //This means the payment is not completed yet.
            }
            return Redirect("/"); ;
        }

        private async Task<bool> CheckIPN(IPNModel model)
        {
            return await CheckoutHelper.IsIPNAuthentic(model);
        }

    }
}

File: NodeJS/Sample Shop Application/SampleShopApplication/api/controllers/homeController.js
'use strict';


var ypco = require('yenepaysdk');

var sellerCode = "0012", //"YOUR_YENEPAY_SELLER_CODE",
    successUrlReturn = "http://localhost:3000/Home/PaymentSuccessReturnUrl", //"YOUR_SUCCESS_URL",
    ipnUrlReturn = "http://localhost:3000/Home/IPNDestination", //"YOUR_IPN_URL",
    cancelUrlReturn = "", //"YOUR_CANCEL_URL",
    failureUrlReturn = "", //"YOUR_FAILURE_URL",
    pdtToken = "PAujBC2Ej1WZaM", // "YOUR_PDT_KEY_HERE",
    useSandbox = true; //false;
    

exports.CheckoutExpress = function(req, res) {
  var merchantOrderId = '12-34'; //"YOUR_UNIQUE_ID_FOR_THIS_ORDER";  //can also be set null
  var expiresInDays = 2; //"NUMBER_OF_DAYS_BEFORE_THE_ORDER_EXPIRES"; //setting null means it never expires
  var checkoutOptions = ypco.checkoutOptions(sellerCode, merchantOrderId, ypco.checkoutType.Express, useSandbox, expiresInDays, successUrlReturn, cancelUrlReturn, ipnUrlReturn, failureUrlReturn);
  var checkoutItem = req.body;
  var url = ypco.checkout.GetCheckoutUrlForExpress(checkoutOptions, checkoutItem);
  res.redirect(url);
};

exports.CheckoutCart = function(req, res) {
  var merchantOrderId = 'AB-CD'; //"YOUR_UNIQUE_ID_FOR_THIS_ORDER";  //can also be set null
  var expiresInDays = 2; //"NUMBER_OF_DAYS_BEFORE_THE_ORDER_EXPIRES"; //setting null means it never expires
  var checkoutOptions = ypco.checkoutOptions(sellerCode, merchantOrderId, ypco.checkoutType.Cart, useSandbox, expiresInDays, successUrlReturn, cancelUrlReturn, ipnUrlReturn, failureUrl);
  var data = req.body;
  var checkoutItems = data.Items;

  //set order level fees like discount, handling fee, tax and delivery fee here
  var totalItemsDeliveryFee = 100;
  var totalItemsDiscount = 50;
  var totalItemsHandlingFee = 30;
  var totalPrice = 0, totalTax1 = 0, totalTax2 = 0;
  checkoutItems.forEach(function(element) {
    totalPrice += element.UnitPrice * element.Quantity;
  });
  totalItemsTax1 = 0.15*totalPrice;
  totalItemsTax2 = 0.02*totalPrice;
  ///////////////////////////////////////////////////////////////

  checkoutOptions.SetOrderFees(totalItemsDeliveryFee, totalItemsDiscount, totalItemsHandlingFee, totalItemsTax1, totalItemsTax2);
  var url = ypco.checkout.GetCheckoutUrlForCart(checkoutOptions, checkoutItems);
  res.json({ "redirectUrl" : url });
};

exports.IPNDestination = function(req, res) {
  var ipnModel = req.body;
  ypco.checkout.IsIPNAuthentic(ipnModel, useSandbox).then((ipnStatus) => {
    //This means the payment is completed
    //You can now mark the order as "Paid" or "Completed" here and start the delivery process
    res.json({"IPN Status": ipnStatus});
  })
  .catch((err) => {
    res.json({ "Error" : err });
  });;
};

exports.PaymentSuccessReturnUrl = function(req, res) {
  var params = req.query;
  var pdtRequestModel = new ypco.pdtRequestModel(pdtToken, params.TransactionId, true);
  ypco.checkout.RequestPDT(pdtRequestModel).then((pdtString) => {
    //This means the payment is completed. 
    //You can extract more information of the transaction from the pdtString
    //You can now mark the order as "Paid" or "Completed" here and start the delivery process
    res.redirect('/');
  })
  .catch((err) => {
    //This means the payment is not completed yet.
    res.redirect('/');
  });;
};
